### PR TITLE
providers/vmware: add support for ovf environments

### PR DESCRIPTION
**Pull request by crawford**
When using VMware's OVF, an environment can optionally be provided
(sometimes done interactively when using tools like vSphere). When this
environment is provided, the guest variables are included in an XML
document at `guestinfo.ovfenv` instead of each of the keys being
directly added to the guest variables.

Still needs docs and testing.

### CODE REVIEW

File: internal/providers/vmware/vmware_amd64.go [hunk 1]
```diff
@@ -56,3 +49,39 @@ func FetchConfig(logger *log.Logger, _ *resource.HttpClient) (types.Config, repo
 	logger.Debug("config successfully fetched")
 	return util.ParseConfig(logger, decodedData)
 }
+
+func fetchRawConfig(logger *log.Logger) (config, error) {
+	info := rpcvmx.NewConfig()
+
+	ovfEnv, err := info.String("ovfenv", "")
+	if err == nil && ovfEnv != "" {
```
Comment (lucab) [inline internal/providers/vmware/vmware_amd64.go]: It looks like this is conflating the error-case and the missing-case. Should this perhaps be split into two err-check+return first and then empty-check+logic blocks?

Comment (crawford) [inline internal/providers/vmware/vmware_amd64.go]: It's probably worth logging any parsing errors. I'll go ahead and split it out.

File: internal/providers/vmware/vmware.go [hunk 2]
```diff
@@ -46,7 +51,7 @@ func decodeData(data string, encoding string) ([]byte, error) {
 		return decodeGzipData(string(gz))
 	}
 
-	return nil, fmt.Errorf("Unsupported encoding %q", encoding)
+	return nil, fmt.Errorf("Unsupported encoding %q", config.encoding)
```
Comment (dgonyeo) [inline internal/providers/vmware/vmware.go]: Tiny nit, shouldn't this be lower case?

**Comment by dgonyeo** [comment]
Uhh, didn't mean to approve this. Aside from Luca's concerns lgtm though.

**Comment by crawford** [comment]
@dgonyeo Luca's concerns should already be addressed.

**Comment by crawford** [comment]
Tested by @arithx on ESXi 6.5. Documentation added.

**Pull request merged**